R/HclustParam.R

```diff
@@ -94,9 +95,14 @@ setMethod("show", "HclustParam", function(object) {
 setMethod("clusterRows", c("ANY", "HclustParam"), function(x, BLUSPARAM, full=FALSE) {
     dargs <- list(quote(as.matrix(x)))
     if (!is.null(BLUSPARAM@metric)) {
-        dargs$metric <- BLUSPARAM@metric
+        dargs$method <- BLUSPARAM@metric
```

Comment:
dargs are passed to dissimilarity function.

dargs had a slot called "metric". Now it is called "method"

--> have you checked that this works with the base::dist function also?

Reply:
Yes, it's even an improvement because before with bluster when indicating the metric, there was an error. Now it should be good for both dist and vegdist

Reply:
I don't know about possible compatibility implications but vegan is using "method", and "metric" is misleading because many dissimilarity measures do not fulfil the definition of a metric.

Reply:
Check this compatibility:

test --> calculate hierarchical clustering with default function (base:dist) --> use other than default metric --> check that result is correct

Reply:
Before I touched the code, when choosing the metric this way: 
`> clusterRows(counts_matrix, HclustParam(metric = "maximum"))`, there was the following error:
`Error in (function (x, method = "euclidean", diag = FALSE, upper = FALSE,  : unused argument (metric = "maximum")`.
So basically, if I'm not mistaken, it wasn't possible to choose the metric because the code passed a unused parameter.
Now that should be fixed. When running the same command, there is no error.
I don't know if that answers everything you asked.

Reply:
weird... both vegdist and dist have the parameter "method" to denote dissmilarity measure... --> this should work as you said